### PR TITLE
Makes Workbase compatible with the latest release of Client Node.js

### PR DESCRIPTION
**Pull request by sorsaffari**
## What is the goal of this PR?
Changes introduced in this PR:
- make Workbase compatible with the latest release of `client-nodejs`
- fix a bug where attributes and roles were not displayed for the selected node in the Schema Designer 
- fix a bug in the scroll behaviour of the right sidebar in the Schema Designer

## What are the changes implemented in this PR?
- make use of the newly introduced `asRemote()` when making Concept API calls
- remove the redundant async behaviour when reading local attributes of a concept
- refactor `mockedConcepts` to reflect the Concept API and make it easier to apply protocol changes moving forward
- remove `buildExplanationQuery` its helpers and associated unit tests as they are no longer used since the `explanation` protocol change
- fix post-processing of `attributes` and `roles` after loading the schema in the Schema Designer
- use flexbox in positioning the remove button for attributes and roles of a selected node in the Schema Designer

### CODE REVIEW

File: src/renderer/components/SchemaDesign/SchemaUtils.js [hunk 1]
```diff
@@ -43,18 +43,21 @@ export async function loadMetaTypeInstances(graknTx) {
 }
 
 // attach attribute labels and data types to each node
-export async function computeAttributes(nodes) {
-  return Promise.all(nodes.map(async (node) => {
-    const attributes = await (await node.attributes()).collect();
+export async function computeAttributes(nodes, graknTx) {
+  const nodesComp = await Promise.all(nodes.map(async (node) => {
```
Comment (adammitchelldev) [inline src/renderer/components/SchemaDesign/SchemaUtils.js:47]: I don't think this is any different from returning the promise directly, what's the thinking here?

File: src/renderer/components/Visualiser/VisualiserUtils.js [hunk 2]
```diff
@@ -68,12 +68,11 @@ export function computeAttributes(nodes, graknTx) {
...
       if (concept.isType()) {
+        attribute.type = concept.label();
         await concept.label().then((label) => { attribute.type = label; });
       } else {
-        await Promise.all([
-          concept.type().then(type => type.label()).then((label) => { attribute.type = label; }),
-          concept.value().then((value) => { attribute.value = value; }),
-        ]);
+        attribute.type = concept.type().label();
+        attribute.value = concept.value();
```
Comment (adammitchelldev) [inline src/renderer/components/Visualiser/VisualiserUtils.js:75]: Nice :)

File: src/renderer/components/shared/CanvasDataBuilder.js [hunk 3]
```diff
@@ -1,6 +1,16 @@
 /* eslint-disable no-await-in-loop */
 import QuerySettings from '../Visualiser/RightBar/SettingsTab/QuerySettings';
 import { META_LABELS, baseTypes } from './SharedUtils';
+import store from '../../store';
+
+const convertToRemote = async (concept) => {
```
Comment (adammitchelldev) [inline src/renderer/components/shared/CanvasDataBuilder.js:6]: I don't think this needs to be `async`, `asRemote` is not async and there are no `await`s.

Comment (sorsaffari) [reply src/renderer/components/shared/CanvasDataBuilder.js:6]: right! I realised that halfway through the changes lol
fixing this ...

Comment (sorsaffari) [reply src/renderer/components/shared/CanvasDataBuilder.js]: resolved in https://github.com/graknlabs/workbase/pull/275/commits/ab3a3c3bf2056939e5765f29692d9c066263f69a

File: src/renderer/components/SchemaDesign/SchemaUtils.js (hunk 1 above)
Comment (sorsaffari) [reply src/renderer/components/SchemaDesign/SchemaUtils.js:47]: ahh, did that way was I was debugging it. resolving this now ...

Comment (sorsaffari) [reply src/renderer/components/SchemaDesign/SchemaUtils.js]: resolved in https://github.com/graknlabs/workbase/pull/275/commits/306a688e895c28da58bbfe0154b9a5bbc71aaece

**Pull request merged**